tests/cpp/test_gpu3.cpp

```diff
@@ -8874,6 +8874,31 @@ TEST_F(NVFuserTest, CpAsyncDataTypeBool) {
   auto cg_outputs = fe.runFusion({t0});
   testValidate(&fusion, cg_outputs, {t0}, __LINE__, __FILE__);
 }
+
+// Intermediate IDs generaetd by rFactor should also remain
+// reductions. See #3327 for more info.
+TEST_F(NVFuserTest, RfactorIntermediateIDs) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(3);
+  fusion.addInput(tv0);
+
+  auto tv1 = sum(tv0, {1, 2});
+  fusion.addOutput(tv1);
+
+  tv1->merge(1, 2);
+  tv1->split(1, 4);
+
+  auto tv2 = tv1->rFactor({-1});
+
+  auto split = dynamic_cast<Split*>(tv2->axis(-1)->definition());
+  ASSERT_NE(split, nullptr);
```

Comment:
Can we also do `EXPECT_TRUE(tv2->axis(-1)->isIteration())` to check that altering the intermediate domains to `Reduction` does not interfere with the factoring; i.e. that this is still only a _partial_ sum.

Reply:
Sure, though it's actually `tv2->axis(-1)->isReduction()` and `tv2->axis(-2)->isIteration` since it's the innermost domain that is rfactored. I'll add an assertion.